src/watchful/client.py

```diff
@@ -1178,14 +1179,14 @@ def exit_backend() -> None:
     """
     This function exits the backend. Note that the API call will usually fail
     because the backend exits before returning a HTTP response so we suppress
-    the error. This is an old API function wrapper and will not be applicable to
-    the dockerized Watchful application instances.
+    the error. This is useful locally, for tests, and during development, but
+    not in dockerized Watchful application instances.
     """
 
     try:
         api("exit")
-    # We allowing passing this error as it is known.
-    except http.client.RemoteDisconnected:
+    # see docstring above
+    except (http.client.RemoteDisconnected, http.client.IncompleteRead):
```

Comment:
When, or in which test cases under which scenarios, does `IncompleteRead` occur?

Reply:
If the remote end goes away before reading any bytes, it will be RemoteDisconnected, after reading HTTP headers (incuding Content-Length) it will be IncompleteRead if fewer bytes are then sent.

Reply:
Which test cases, at which point in the test suite, or at what test conditions, are you meeting IncompleteRead that you regard as expected and therefore not raising its exception?

I think it's important we raise the exception, save areas that we know it will happen in because of known reasons that we accept, because we should almost always expect not having IncompleteRead while testing.

This ensures unexpected errors don't get shielded from the radar.

Reply:
Did you notice that we are in the context of the `exit_backend()` method here?

```
This function exits the backend. Note that the API call will usually fail
because the backend exits before returning a HTTP response so we suppress
the error.
```

The user sends a request to the backend that says "end the process and free all resources" and the backend frees all resources instantly. The backend will not spend any more resources on any outstanding HTTP requests, *including this one*. The user would never possibly care about or want to see these errors in any possible scenario, so we suppress them.

Reply:
Hey, I'm going to merge this as the other PR is also landing today. I'm not sure my explanation was sufficient here, but let's find a time to chat about this!